providers/openshift/provider.go

```diff
@@ -145,7 +153,16 @@ func (p *OpenShiftProvider) newOpenShiftClient() (*http.Client, error) {
 
 	// try to retrieve a cached client
 	metadataHash, err := util.GetFilesMetadataHash(capaths)
-	if httpClient, ok := p.httpClientCache.Load(metadataHash); ok {
+	if err != nil {
+		return nil, err
+	}
+
+	oauthServerCert, err := p.configMapLister.ConfigMaps("openshift-config-managed").Get("oauth-serving-cert")
```

Comment:
probably obvious, but this assumes that the client `newOpenShiftClient` is never ever reused betwen invocations, correct?

Reply:
the `newOpenShiftClient` does in fact promote the reuse of the constructed client (which is cached in the `httpClientCache`)

Reply:
ah, thank you, I wasn't aware. How do we deal with changes/rotation on those certs then?

Reply:
I just noticed there was a bug that would cause failure of the reuse, that's now fixed (see `cachedKey`).

When the certs change, we cache a new client that's then used. Each of the calls to OpenShift usually starts by calling the `newOpenshiftClient()`

Reply:
>  Each of the calls to OpenShift usually starts by calling the newOpenshiftClient()

ah, combined with the cached http client that makes sense to me now :+1: